mm2src/mm2_rpc/src/data/legacy/orders.rs

```diff
@@ -95,106 +197,309 @@ impl Default for MatchBy {
-    pub total_bids_rel: TotalBidsRelVol,
+    pub price: MmNumber,
+    #[serde(default)]
+    pub max: bool,
+    #[serde(default)]
+    pub volume: MmNumber,
+    pub min_volume: Option<MmNumber>,
+    #[serde(default = "true_f")]
+    pub cancel_previous: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub base_confs: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub base_nota: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rel_confs: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rel_nota: Option<bool>,
+    #[serde(default = "true_f")]
+    pub save_in_history: bool,
 }
 
-construct_detailed!(TotalAsksBaseVol, total_asks_base_vol);
-construct_detailed!(TotalAsksRelVol, total_asks_rel_vol);
-construct_detailed!(TotalBidsBaseVol, total_bids_base_vol);
-construct_detailed!(TotalBidsRelVol, total_bids_rel_vol);
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(tag = "method", rename = "orderbook_depth")]
+pub struct OrderbookDepthRequest {
+    pub pairs: Vec<(String, String)>,
+}
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct RpcOrderbookEntry {
-    pub coin: String,
-    pub address: String,
-    pub price: BigDecimal,
-    pub price_rat: BigRational,
-    pub price_fraction: Fraction,
-    #[serde(rename = "maxvolume")]
-    pub max_volume: BigDecimal,
-    pub max_volume_rat: BigRational,
-    pub max_volume_fraction: Fraction,
-    pub min_volume: BigDecimal,
-    pub min_volume_rat: BigRational,
-    pub min_volume_fraction: Fraction,
-    pub pubkey: String,
-    pub age: u64,
-    pub uuid: Uuid,
-    pub is_mine: bool,
-    #[serde(flatten)]
-    pub base_max_volume: DetailedBaseMaxVolume,
-    #[serde(flatten)]
-    pub base_min_volume: DetailedBaseMinVolume,
-    #[serde(flatten)]
-    pub rel_max_volume: DetailedRelMaxVolume,
-    #[serde(flatten)]
-    pub rel_min_volume: DetailedRelMinVolume,
-    #[serde(flatten)]
-    pub conf_settings: Option<OrderConfirmationsSettings>,
+#[derive(Debug, Deserialize, Serialize)]
+pub struct PairWithDepth {
+    pub pair: (String, String),
+    pub depth: PairDepth,
 }
 
-construct_detailed!(DetailedBaseMaxVolume, base_max_volume);
-construct_detailed!(DetailedBaseMinVolume, base_min_volume);
-construct_detailed!(DetailedRelMaxVolume, rel_max_volume);
-construct_detailed!(DetailedRelMinVolume, rel_min_volume);
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct PairDepth {
+    pub asks: usize,
+    pub bids: usize,
+}
 
 #[derive(Debug, Serialize, Deserialize)]
-pub struct AggregatedOrderbookEntry {
-    #[serde(flatten)]
-    pub entry: RpcOrderbookEntry,
-    #[serde(flatten)]
-    pub base_max_volume_aggr: AggregatedBaseVol,
-    #[serde(flatten)]
-    pub rel_max_volume_aggr: AggregatedRelVol,
+#[serde(tag = "method", rename = "orders_history_by_filter")]
+pub struct OrdersHistoryRequest {
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
So for every request struct (new or old) we have to add `#[serde(skip_serializing_if = "Option::is_none")]` for all optional parameters so the that it can work in the CLI. Can this be done in an easier way to avoid coupling between cli code and komoDeFi code?

Reply:
[introduced serde_with::skip_serializing_none](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1932/commits/001abfffbf55e7e82e4c1ce20b9b0083c528d780)

Reply:
Introduction of `serde_with` introduced a duplicated `strsim` version. It's alright by me since the 0.8.0 version is used by solana which is an unused feature but I would like to get @ozkanonur opinion on this.

Reply:
I would not recommend updating the internal types of mm2 in such an aggressive manner, especially when it comes to deserializations. Clients heavily rely on these type formats.

Regarding the issue of dependency duplication, by roughly thinking, an alternative solution could be creating wrapper types in the adex_cli. This would allow you to implement the required changes in a more controlled manner and avoid doing changes on mm2 internals.

For example:

```rs
use serde::{ser::SerializeStruct, Serialize, Serializer};
use serde_derive::Deserialize;

#[derive(Serialize, Deserialize, Clone)]
struct Person {
    name: String,
    surname: Option<String>,
}

#[derive(Deserialize)]
struct PersonContainer {
    #[serde(flatten)]
    person: Person,
}

impl Serialize for PersonContainer {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: Serializer,
    {
        let mut state = serializer.serialize_struct("PersonContainer", 2)?;

        state.serialize_field("name", &self.person.name)?;

        if let Some(surname) = &self.person.surname {
            state.serialize_field("surname", surname)?;
        }

        state.end()
    }
}

fn main() {
    let x = Person {
        name: "Onur".to_string(),
        surname: None,
    };

    let y = PersonContainer { person: x.clone() };

    let serialized = serde_json::to_string(&x).unwrap();

    assert_eq!(serialized, r#"{"name":"Onur","surname":null}"#);

    let serialized2 = serde_json::to_string(&y).unwrap();
    assert_eq!(serialized2, r#"{"name":"Onur"}"#);
}
```

Reply:
> Another solution can be to port [skip_serializing_none](https://github.com/jonasbb/serde_with/blob/eb48aee4a647b423c69ebf87c8ffeac2f0f77047/serde_with_macros/src/lib.rs#L319) code to [`mm2src/derives`](https://github.com/KomodoPlatform/komodo-defi-framework/tree/f7cd5db3ec96648391cfc3d98620769a795df0c8/mm2src/derives)

This might require some maintanince work on major serde bumps

Reply:
[done](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1932/commits/b703f4ba40ad69c05c76ad3de2a39d749b7c82d9)

Reply:
Thank you, it's much more better place 
[done](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1932/commits/3c96319e933119f17b847a3df3b66392c92c4de7)